File: src/main/java/cn/pompip/myblog/interceptor/SecurityInterceptor.java
package cn.pompip.myblog.interceptor;

import org.springframework.stereotype.Component;
import org.springframework.web.servlet.HandlerInterceptor;
import org.springframework.web.servlet.ModelAndView;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

@Component
public class SecurityInterceptor implements HandlerInterceptor {
    @Override
    public  boolean preHandle(HttpServletRequest request,HttpServletResponse response,Object handler )throws Exception  {
        if (request.getSession().getAttribute("user") == null) {
            response.sendRedirect("/user/login");
            return false;
        } else {
            return true;
        }
    }

    @Override
    public void postHandle(HttpServletRequest request , HttpServletResponse response, Object handler , ModelAndView modelAndView)  {

    }
}

File: src/main/kotlin/cn/pompip/myblog/interceptor/SecurityInterceptor.kt
package cn.pompip.myblog.interceptor

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Configuration
import org.springframework.stereotype.Component
import org.springframework.web.servlet.HandlerInterceptor
import org.springframework.web.servlet.ModelAndView
import org.springframework.web.servlet.config.annotation.InterceptorRegistry
import org.springframework.web.servlet.config.annotation.ViewControllerRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@Configuration
class SecurityConfig : WebMvcConfigurer {
    @Autowired
    lateinit var interceptor: SecurityInterceptor;

    override fun addViewControllers(registry: ViewControllerRegistry) {
        super.addViewControllers(registry)
    }

    override fun addInterceptors(registry: InterceptorRegistry) {
        registry.addInterceptor(interceptor).addPathPatterns("/markdown").addPathPatterns("/archives")
    }
}

@Component
class SecurityInterceptor : HandlerInterceptor {
    override fun preHandle(request: HttpServletRequest, response: HttpServletResponse, handler: Any): Boolean {
        if (request.session.getAttribute("user") == null) {
            response.sendRedirect("/user/login")
            return false
        } else {
            return true
        }
    }

    override fun postHandle(request: HttpServletRequest, response: HttpServletResponse, handler: Any, modelAndView: ModelAndView?) {
        super.postHandle(request, response, handler, modelAndView)
    }


}